### ISSUE TITLE
False positive: `fill-(--color)/(--opacity)` flagged as "Use the named theme token"

### ISSUE THREAD

**Issue by nbazille-ipsos** [sergioazoc/oxlint-tailwindcss#125]
Summary

The plugin flags the class `fill-(--color)/(--opacity)` with the suggestion:

"`fill-(--color)/(--opacity)` can be written as `fill-color/(--opacity)`. Use the named theme token"

This appears to be a false positive: the class using CSS variables for the color and the opacity at the same time is valid Tailwind usage in our projects and should not be rewritten to a named theme token.

Reproduction

1. Add a class that uses CSS variables in the fill utility, for example:

```html
<svg class="fill-(--color)/(--opacity)">...</svg>
```

2. Run oxlint with this plugin on the file.
3. Observe the diagnostic above recommending conversion to a named theme token.

Expected behavior

- The rule should accept arbitrary values that use CSS custom properties (CSS variables) inside the fill utility and should not suggest rewriting them to a named theme token.
- If the rule intends to detect cases that can be replaced by named theme tokens, it should ignore values that are CSS variables (start with `--`) or otherwise clearly arbitrary values.

Note that the issue happens only with `fill-(--color)/(--opacity)` but not with `fill-(--color)`

**Comment by sergioazoc** [comment]
Fixed and shipped in v1.10.1. Thanks for the report @nbazille-ipsos, this was a good one.

The root of it: `prefer-theme-tokens` was checking the replacement candidate with the opacity modifier still glued on (`fill-color/(--opacity)`), and the validity check is tolerant, anything with parens or brackets in it reads as a valid arbitrary value. So the `(--opacity)` part alone was enough to make `fill-color/…` look real, even though `fill-color` is not a token. The rule then "fixed" a perfectly valid class into one that compiles to nothing, silently dropping both the color and the opacity. On an autofix rule that is about as bad as it gets.

Digging in, `/(--opacity)` was just the shape you happened to hit. The same destructive rewrite fired for `fill-(--color)/[0.8]`, `fill-(--color)/[50%]`, and the bracket color forms too.

The fix decides from the base token (`fill-color`) instead of the full string with the modifier, so the outcome no longer depends on the modifier's shape, and the modifier rides along untouched. All of those cases are covered now.

`npm i -D oxlint-tailwindcss@1.10.1` and you should be good. Let me know if anything still looks off.
